public/src/pages/views/MemberView.tsx

```diff
@@ -158,12 +134,39 @@ export class MemberView extends React.Component<IUserViewerProps, IUserViewerSta
             `Warning! This action is irreversible!
             Do you want to reject the student?`,
         )) {
-            const result =
-             await this.props.courseMan.changeUserState(userRel.link, Enrollment.UserStatus.REJECTED);
-            if (result) {
-                userRel.link.setStatus(Enrollment.UserStatus.REJECTED);
+            let readyToDelete =
+             await this.props.courseMan.isEmptyRepo(this.props.course.getId(), userRel.user.getId(), 0);
+            if (!readyToDelete) {
+                readyToDelete = confirm(
+                    `Warning! User repository is not empty.
+                    Do you still want to reject the user?`
+                    );
             }
-        }
+
+            if (readyToDelete) {
+                const result =
+             await this.props.courseMan.changeUserState(userRel.link, Enrollment.UserStatus.NONE);
+                if (result) {
+                    switch (userRel.link.getStatus()) {
+                        case Enrollment.UserStatus.PENDING:
+                            const i = this.state.pendingUsers.indexOf(userRel);
+                            if (i >= 0) {
+                            this.state.pendingUsers.splice(i, 1);
```

Comment:
There are some indentation problems with this method. Perhaps run the linter and typescript formater in vscode?

Reply:
My linter fails to see any problem, could you specify?

Reply:
The linter only tells me
- Missing trailing comma [142,59]
- block is empty [227,23]

But the problem I'm actually referring to on line 148, 154, 160 and 169 (and several others) is that the line indentations are not as expected. This should normally be fixed by saving the file, I think; at least my vscode is set up to format typescript on save.

Reply:
Yeah, trailing comma is fixed, and the empty block is a legacy part of passing a method between elements that I rather would not touch right now.
But indentation problems are still invisible for my linter. Found and fixed some by hand.